src/http/RequestMethod.h

```diff
@@ -28,15 +28,12 @@ class HttpRequestMethod : public RefCountable
 public:
     HttpRequestMethod() : theMethod(Http::METHOD_NONE), theImage() {}
     HttpRequestMethod(Http::MethodType const aMethod) : theMethod(aMethod), theImage() {}
+    HttpRequestMethod(const HttpRequestMethod &) = default;
     explicit HttpRequestMethod(const SBuf &);
 
     void HttpRequestMethodXXX(char const *); // deprecated old c-string to SBuf converter.
 
-    HttpRequestMethod & operator = (const HttpRequestMethod& aMethod) {
-        theMethod = aMethod.theMethod;
-        theImage = aMethod.theImage;
-        return *this;
-    }
+    HttpRequestMethod & operator = (const HttpRequestMethod& aMethod) = default;
```

Comment:
> Kinkie: The customized assignment operator is like the default

It does look like that, but those looks are deceiving. Unlike the default, the custom operator does not copy the Lock::count_ field inherited via RefCountable. If the current PR implementation is applied to a dynamically-allocated HttpRequestMethod object, then Squid will either crash or leak (depending on the difference between `this->count_` and `aMethod.count_` values).

Most likely, you do not want to fix the bigger problem with Lock copying in this PR. If you agree, you may want to undo this buggy change and add an explicit `do not copy the Lock::count_!` comment to the operator implementation.

Reply:
Yes, I'll do that. Thanks for catching this